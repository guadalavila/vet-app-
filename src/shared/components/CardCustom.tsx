import React from 'react';
import { StyleSheet, View } from 'react-native';
import { colors } from '../utils/colors';
import { size } from '../utils/size';
import { GlobalStyles } from '../utils/styles';
import CustomText from './CustomText';
import { typography } from '../utils/typography';

interface ICardCustomProps {
    title: string;
    value: string;
    valueExtra?: string;
    childExtra?: React.ReactNode;
}

const CardCustom = ({ title, value, valueExtra, childExtra }: ICardCustomProps) => {
    return (
        <View style={styles.container}>
            <View style={GlobalStyles.rowBetween}>
                <CustomText style={styles.title}>{title}</CustomText>
                {childExtra}
            </View>
            <CustomText style={[styles.value]}>
                {value}
                <CustomText style={styles.valueExtra}>{valueExtra}</CustomText>
            </CustomText>
        </View>
    );
};

export default CardCustom;

const styles = StyleSheet.create({
    container: {
        width: '30%',
        backgroundColor: colors.light.secondary,
        borderRadius: 10,
        padding: size.L,
        margin: size.S,
    },
    title: {
        fontWeight: '600',
        color: colors.light.grey,
    },
    value: {
        marginVertical: size.S,
        color: colors.light.white,
        fontSize: typography.size.M,
        fontWeight: 'bold',
    },
    valueExtra: {
        fontSize: typography.size.S,
        color: colors.light.greySecondary,
    },
});
