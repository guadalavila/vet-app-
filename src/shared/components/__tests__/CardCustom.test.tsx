import { render, screen } from '@testing-library/react-native';
import React from 'react';
import CardCustom from '../CardCustom';

describe('CardCustom', () => {
    test('should match snapshot', () => {
        const container = render(<CardCustom title='Sexo' value='Macho' />);
        expect(container).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<CardCustom title='Sexo' value='Macho' />);
        expect(screen.getByTestId('cardCustom')).toBeTruthy();
    });

    test('should render title', () => {
        render(<CardCustom title='Sexo' value='Macho' valueExtra='Extra' />);
        expect(screen.getByText('Sexo')).toBeTruthy();
        expect(screen.getByText('Extra')).toBeTruthy();
    });
});
