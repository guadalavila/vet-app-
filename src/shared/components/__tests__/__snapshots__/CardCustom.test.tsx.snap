// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardCustom should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2A2F4F",
      "borderRadius": 10,
      "margin": 4,
      "padding": 8,
      "width": "30%",
    }
  }
  testID="cardCustom"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#11100F",
            "fontSize": 16,
          },
          Object {
            "color": "#EEEEEE",
            "fontWeight": "600",
          },
        ]
      }
    >
      Sexo
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginVertical": 4,
          },
        ],
      ]
    }
  >
    Macho
    <Text
      style={
        Array [
          Object {
            "color": "#11100F",
            "fontSize": 16,
          },
          Object {
            "color": "#f5f5f5",
            "fontSize": 16,
          },
        ]
      }
    />
  </Text>
</View>
`;
